partner_compassion/models/parter_segment_affinity.py

```diff
@@ -0,0 +1,135 @@
+        [-5.06030329552021, -3.60413990223956, -1.04375649300268, -4.47657656335253, 0],
+        [-1.66377418504539, -5.32573310317821, -2.44074881077657, -4.50674881857412, 0],
+        [-4.77259566923145, -3.15598279721272, -3.52425639320478, -6.67368137733622, 0],
+        [0.880182155165077, -1.44671948132195, -2.4884760610343, -1.02472137413719, 0],
+        [-0.0326103888365477, -2.99571602280864, -2.69104850676188, -1.75084871433893, 0],
+        [-0.783439038458744, 0.917536337079493, 0.86575431420832, 2.93637326365812, 0],
+        [-0.566278267023883, -0.893538321476831, -3.80511027210482, -4.14668710742975, 0],
+        [2.86138579763347, 3.3103473304014, 12.7709974491643, 3.46942313908716, 0],
+        [-0.404907139047755, -2.55346578831099, 13.4855455716769, 1.55681107187723, 0]
+    ]
+
+    answer_constant_engine = [
+        -6.7614377947751, -8.85904330388748, -5.02071420157718, -6.23710433298537, 0
+    ]
+
+    @api.model
+    def create(self, vals):
+        """
+        Create a segment_affinity instance. Update partner reference on segment affinity
+        :param vals: values used for model creation
+        :return: id of the newly created model
+        """
+        seg_affin_id = super(ResPartnerSegmentAffinity, self.sudo()).create(vals)
+
+        # update partner.all_segment_affinity will trigger primary and secondary segment computation
+        # seg_affin_id.partner_id.write({'all_segments_affinity': [(4, seg_affin_id.id)]})
+        seg_affin_id.partner_id.all_segments_affinity = [(4, seg_affin_id.id)]
+        # seg_affin_id.partner_id._compute_prim_sec_segments()
+        return seg_affin_id
+
+    @api.multi
+    def unlink(self):
+        """
+        Unlink insance of the model. Update partner reference.
+        :return: True
+        """
+        for seg_affin in self:
+            seg_affin.partner_id.all_segments_affinity = [(2, seg_affin.id)]
+            super(ResPartnerSegmentAffinity, seg_affin).unlink()
+        return True
+
+    @api.model
+    def segment_affinity_engine(self, answer_as_array, partner_id):
+        """
+        Adaptation of the segmentation engine used by GMC. Output the segmentation affinity for each of the 5 segment
+        describe by GMC. Creating all the segment affinity instance will update the partner primary and secondary
+        segment.
+
+        :param answer_as_array: A well formatted list of user_input_line quizz_mark
+        (see partner_compassion.survey)
+        :param partner_id: the partner related to the answers
+        :return: True (update partner all_segments_affinity)
+        """
+
+        # simplified explanation of the segmentation percentage computation
+        # answer_array * matrix -> scaled matrix ( + constant) -> segmentation score
+        # -> segmentation percentage (= segmentation affinity)
+
+        vals = {
+            'partner_id': partner_id.id
```

Comment:
You only use `partner_id.id` so I would suggest you directly pass this as parameter when you call the method and here you just use it like this.
```suggestion
            'partner_id': partner_id
```